openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -544,8 +564,16 @@ def get(self, request, usage_key_string, *args, **kwargs):  # lint-amnesty, pyli
         view = STUDENT_VIEW
         if request.user.is_anonymous:
             view = PUBLIC_VIEW
+        sequence_data = sequence.get_metadata(view=view)
+
+        # Insert usage_key_hashes of this sequence and its units into the response.
+        # We do this here in order to avoid further complicating seq_module
+        # with details of usage key hashing.
+        sequence_data["usage_key_hash"] = learning_sequence.usage_key_hash
+        for unit_data in sequence_data["items"]:
+            unit_data["usage_key_hash"] = hash_usage_key(unit_data["item_id"])
 
-        return Response(sequence.get_metadata(view=view))
+        return Response(sequence_data)
```

Comment:
New behavior 2: The Sequence Metadata API will include hashes of the sequence usage_key and all its unit usage_keys.

This way, the frontend can construct a `unit_usage_key<->unit_usage_key_hash` mapping as soon as this API call returns.